### DISCUSSION TITLE
Add `$collection->toDot();` method to transform the collection to the dotted notation.

### DISCUSSION THREAD

**Discussion by mikevandiepen** [laravel/framework#48036] [Ideas] [answered]
At my company we are in need for this function, the `->undot()` method already exists but the `->todot()` method doesnt.
For now I use a macro but it would be nice to see it implemented.

My approach:
```PHP
Collection::macro('toDot', function (): Collection {
    /* @var $this Collection taking the identical fqcn for later reconstruction, otherwise it might mess with custom collections */ 
    $collection = get_class($this);

    // Prefixing the keys of the array with an underscore, otherwise the keys will be lost and mapped to an chronological default order
    // possibly breaking the collection content for certain scenarios.
    $prepend = '_';

    return $collection
        ::make(Arr::dot($this, $prepend))
        ->mapWithKeys(function ($value, string $dottedKey) use ($prepend): array {
            return [
                substr($dottedKey, strlen($prepend)) => $value,
            ];
        });
});
```

**Comment by henzeb** [comment, marked as answer]
I'd call it dot and not toDot. Unfortunately, that method already exists: https://laravel.com/docs/10.x/collections#method-dot

**Reply by mikevandiepen** [reply]
Thanks, did not know about that collection method, we are using an older version of Laravel (for now) so that is not a solution yet. Looking at how its done it still has the flaw that it re-indexes numeric keys so for our use-case it is not a solution. But thats on us :)
